src/components/HistoryTable/historyTableContextReducer.ts

```diff
@@ -122,6 +127,22 @@ export const reducer = (state: HistoryTableReducerState, action: Action) => {
         ...state,
         columnLimit: action.limit,
       };
+    case "onChangeTableWidth":
+      if (calcColumnLimitFromWidth(action.width) === state.columnLimit) {
+        return state;
+      }
+      return {
+        ...state,
+        visibleColumns: state.columns.slice(
+          0,
+          calcColumnLimitFromWidth(action.width)
+        ),
+        columnLimit: calcColumnLimitFromWidth(action.width),
+        currentPage: 0,
+        pageCount: Math.ceil(
+          state.columns.length / calcColumnLimitFromWidth(action.width)
```

Comment:
[nit] You could save `calcColumnLimitFromWidth(action.width)` as a variable since it's used a few times.

Reply:
I would have to move the variable outside of the switch statement because of the rule [no-case-declarations](https://eslint.org/docs/rules/no-case-declarations). I think a neater way to do this would be to create a separate function that performs the logic in the case by accepting a state and action and returning the state but we haven't really established that pattern for the rest of our reducer cases.

Reply:
You would actually just need to add a `{` to the case declaration. We do something similar on line 111